### PR TITLE
docs(request-transformer) document missing features in plugin

### PR DESCRIPTION
**Pull request by kikito**

### CODE REVIEW

File: app/_hub/kong-inc/request-transformer/index.md [hunk 1]
```diff
@@ -3,9 +3,9 @@ name: Request Transformer
 publisher: Kong Inc.
 version: 1.2.5
 
-desc: Modify the request before hitting the upstream server
+desc: Use regular expressions, variables and templates to transform requests
 description: |
-  Transform the request sent by a client on the fly on Kong, before hitting the upstream server.
+  The Request Transformer plugin for Kong allows simple transforming the requests before they reach the upstream server. These transformations can be simple substitutions or complex ones matching portions of incoming requests using regular expressions, save those matched strings into variables, and substitute those strings into transformed requests via flexible templates.
```
Comment (tlblessing) [inline app/_hub/kong-inc/request-transformer/index.md:8]:
```suggestion
  The Request Transformer plugin for Kong allows simple transformation of requests before they reach the Upstream server. These transformations can be simple substitutions or complex ones matching portions of incoming requests using regular expressions, saving those matched strings into variables, and substituting those strings into transformed requests using flexible templates.
```

Comment (kikito) [reply app/_hub/kong-inc/request-transformer/index.md:8]: Took all the suggestions here except "Upstream server" - in this case we are using "upstream" as an adjective. And yes, this is more confusing than it needs to be. I personally think we should have named our Upstreams differently.

File: app/_hub/kong-inc/request-transformer/index.md [hunk 2]
```diff
@@ -75,6 +75,13 @@ params:
       required: false
       value_in_examples: [ "formparam-toremove", "formparam-another-one" ]
       description: List of parameter names. Remove the parameter if and only if content-type is one the following [`application/json`, `multipart/form-data`,  `application/x-www-form-urlencoded`] and parameter is present.
+    - name: replace.uri
+      required: false
+      description: Updates the upstream request URI with given value. This value can only be used to update the path part of the URI, not the scheme, nor the hostname.
```
[the remark this answers is not in the document]
Comment (kikito) [reply app/_hub/kong-inc/request-transformer/index.md:80]: Took everything except "Upstream request". Same reason as above.

File: app/_hub/kong-inc/request-transformer/index.md [hunk 3]
```diff
@@ -124,25 +131,110 @@ params:
...
+```
+
+Now send a request without setting header `x-user-id`
+
+```bash
+$ curl -i -X GET localhost:8000/requests/user/foo
+```
+
+Plugin will add a new header `x-consumer-id` with value alice before proxying
+request upstream. Now try sending request with header `x-user-id` set
```
[the remark this answers is not in the document]
Comment (kikito) [reply app/_hub/kong-inc/request-transformer/index.md:223]: Upstream vs upstream again.

[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/request-transformer/index.md:223]: @kikito  docs convention was to uppercase terms that appeared in the glossary that used to be alongside the sidebar. I'm working on the Reedelk transformer plugin and that has both upstream and downstream (down not in glossary), and so I think it is an odd convention that should be dropped.

File: app/_hub/kong-inc/request-transformer/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/request-transformer/index.md]: @kikito  what reason above?

**Comment by tlblessing** [comment]
Direct review link:

https://deploy-preview-2186--kongdocs.netlify.app/hub/kong-inc/request-transformer/

@kikito in the non-generated portion of this doc, I'm seeing a lot of with/without db entires that could use some div tags and take up less real estate. I'll take care of that.

**Pull request merged**